webapp/channels/src/components/admin_console/multiselect_settings.tsx

```diff
@@ -26,67 +25,61 @@ interface Props {
     noResultText?: React.ReactNode;
 }
 
-interface State {
-    error: boolean;
-}
-
-export default class MultiSelectSetting extends React.PureComponent<
-Props,
-State
-> {
-    static defaultProps: Partial<Props> = {
-        disabled: false,
-    };
-
-    constructor(props: Props) {
-        super(props);
+const MultiSelectSetting: React.FC<Props> = ({
+    id,
+    values,
+    label,
+    selected,
+    onChange,
+    disabled = false,
+    setByEnv,
+    helpText,
+    noResultText,
+}) => {
+    const [error, setError] = useState(false);
 
-        this.state = {error: false};
-    }
+    const handleChange = useCallback((newValue: ValueType<Option>) => {
+        const updatedValues = newValue ? (newValue as Option[]).map((n) => 
+        n.value) : [];
```

Comment:
Not sure if an artifact of the github visualization, or some linter trying to do things different, but... the line breaks seem off. I would leave it as they used to be before, but I am happy to consider other options.

Reply:
updates